chemprop/uncertainty/calibrator.py

```diff
@@ -1,21 +1,61 @@
-from abc import abstractmethod
+from abc import ABC, abstractmethod
+from typing import Self
 
 from torch import Tensor
 
 from chemprop.utils.registry import ClassRegistry
 
 
-class UncertaintyCalibrator:
+class UncertaintyCalibrator(ABC):
+    """
+    A class for calibrating the predicted uncertainties.
+    """
+
     @abstractmethod
-    def fit(self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor) -> None:
+    def fit(self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor) -> Self:
         """
         Fit calibration method for the calibration data.
+
+        .. note::
+            The `preds` is only needed for regression tasks. The `uncs` would be the predicted variance for regression tasks,
+            the predicted probability of class 1 for binary classification and the probabilities for each class for multiclass classification.
```

Comment:
It would read better as follows:

The 'preds' variable is required only for regression tasks.
The 'uncs' variable holds different values depending on the task:
- For regression tasks, 'uncs' represents the predicted variance.
- For binary classification, 'uncs' is the predicted probability of class 1.
- For multiclass classification, 'uncs' contains the predicted probabilities for each class.